staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -190,19 +229,11 @@ type Backoff struct {
 // If the condition never returns true, ErrWaitTimeout is returned. All other
 // errors terminate immediately.
 func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
-	duration := backoff.Duration
-	for i := 0; i < backoff.Steps; i++ {
-		if i != 0 {
-			adjusted := duration
-			if backoff.Jitter > 0.0 {
-				adjusted = Jitter(duration, backoff.Jitter)
-			}
-			time.Sleep(adjusted)
-			duration = time.Duration(float64(duration) * backoff.Factor)
-		}
+	for backoff.Steps > 0 {
 		if ok, err := condition(); err != nil || ok {
 			return err
 		}
+		time.Sleep(backoff.Step())
```

Comment:
this should be:
```
delay := backoff.Step()
if backoff.Steps > 0 {
  time.Sleep(delay)
}
```

otherwise we sleep for no reason (and simply delay returning ErrWaitTimeout

Reply:
Updated slightly differently